lightning/src/chain/onchaintx.rs

```diff
@@ -382,7 +382,39 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 		where F::Target: FeeEstimator,
 					L::Target: Logger,
 	{
-		if cached_request.outpoints().len() == 0 { return None } // But don't prune pending claiming request yet, we may have to resurrect HTLCs
+		let request_outpoints = cached_request.outpoints();
+		if request_outpoints.is_empty() {
+			// Don't prune pending claiming request yet, we may have to resurrect HTLCs. Untractable
+			// packages cannot be aggregated and will never be split, so we cannot end up with an
+			// empty claim.
+			debug_assert!(cached_request.is_malleable());
+			return None;
+		}
+		// If we've seen transaction inclusion in the chain for our request, we don't need to
+		// continue generating more claims. We'll keep tracking the request to fully remove it once
+		// it reaches the confirmation threshold, or to generate a new claim if the transaction is
+		// reorged out.
+		if self.onchain_events_awaiting_threshold_conf.iter()
+			.any(|event_entry| if let OnchainEvent::Claim { claim_request } = event_entry.event {
+				for outpoint in &request_outpoints {
+					if let Some(first_claim_txid_height) = self.claimable_outpoints.get(outpoint) {
+						if first_claim_txid_height.0 != claim_request {
+							// The request's outpoint spend does not correspond to the current claim event.
+							return false;
+						}
+					} else {
+						// The request's outpoint spend does not exist yet.
+						return false;
+					}
+				}
+				// All of the request's outpoints were spent by the claim event.
+				true
+			} else {
+				false
```

Comment:
nit: "The onchain event is not a claim. Keep seeking if we have a request containing all the outpoints".